### DISCUSSION TITLE
Ebesucher proxy detection

### DISCUSSION THREAD

**Discussion by x11r6** [engageub/InternetIncome#52] [General]
First of all, thanks to the author for its amazing job!
I tried to get Ebesucher to work with proxies in several ways, but I always failed. If I install Linux on a VMWare virtual machine, configure tun2socks to route all the traffic throuh my socks i always get upaid paged.
When using your solution it seems to be working fine (even tho i keep having to click the "Surf now!" button every hour or so.

How is this even possible? How can Ebesucher detect that im using a proxy on my VM, but somehow it is ok to use it on your docker container?

**Comment by engageub** [comment]
Hi,
Could you please let me know what is meant by "i always get upaid paged".
Are you seeing the proxy address  when using https://whatismyipaddress.com in firefox browser ?
Many have received payouts from this. We are using browser with the official extension. 
There are many extensions to use proxy within firefox itself. They need it for marketing research and testing, so it does not matter if you look at the browser or not as long the surfing continues. 

Thank you

**Comment by x11r6** [comment]
Unpaid pages are buffer pages that Ebesucher shows when it doesn't like the quality of the traffic you generate (like in the image). When you see these pages you get 0 BTP, so youre not earning anything.
I also got many payout from ebesucher, but whenever i tried to use the surfbar with a proxy i always got unpaid pages.
Now that i'm using your solution it somehow works even by using proxies, and i'm wondering how, that's it 😄 
If I run an Ubuntu virtual machine, configure tun2socks and run Ebesucher i always get unpaid content. If i run https://whatismyipaddress.com/ into the virtual machine i see my proxy ip address, so im sure im behind proxy.
![image](https://github.com/engageub/InternetIncome/assets/25671232/44f12d6d-ffb4-4f11-84ca-3a49f3a22ffc)

**Reply by engageub** [reply]
Thank you for the clarification. Yes, that's the problem which usually happens when non-residential proxy is used. It redirects to unpaid pages most of the time and sometimes paid pages also are shown. But the earnings are usually good with residential proxies. 
The code is transparent, you can look into the code anytime to understand how it works. 

Thank you

**Comment by engageub** [comment]
The code for recaptcha solver and click surf are already deployed to test branch. This works only in chrome browser as of now. 
https://github.com/engageub/InternetIncome/tree/test

Thank you

**Reply by engageub** [reply]
Hi,
Firefox in main branch now has recaptcha solver included.
Thank you
